packages/client/lib/service/txpool.ts

```diff
@@ -282,6 +282,10 @@ export class TxPool {
         `Tx gaslimit of ${tx.gasLimit} exceeds block gas limit of ${block.gasLimit} (exceeds last block gas limit)`
       )
     }
+
+    const vmCopy = await this.vm.copy()
```

Comment:
i don't think we need copy here, txpool's vm is already a copy and its only being used in validate where the stateroot is already being set

Reply:
It's just using the execution module's VM as [here](https://github.com/ethereumjs/ethereumjs-monorepo/blob/0b569e67daf9275d5e34a6674ab7a3afb4117296/packages/client/lib/service/txpool.ts#L153) so I believe it's all the same instance across the whole client.

Reply:
ok, will review but feels like vmExecution is running on its own copy, but for the sake of the PR, we can copy vm in the constructor itself

Reply:
ok, will review but feels like vmExecution is running on its own copy, but for the sake of the PR, we can copy vm in the constructor itself

Reply:
Wouldn't you end up operating on out of sync state manager if you do that since the txpool's state manager wouldn't get updated when the VM runs blocks?

Reply:
that won't be the issue as setStateRoot will set the correct one, however i do see that the VMs are shared between txpool and execution, but i am wondering why would this vm's state not being updated, but i am ok with the copy here :+1: